sshd-common/src/main/java/org/apache/sshd/client/config/hosts/HostConfigEntry.java

```diff
@@ -1087,6 +1087,7 @@ public static void writeHostConfigEntries(
      */
     public static List<String> parseConfigValue(String value) {
         String s = GenericUtils.replaceWhitespaceAndTrim(value);
+        s = GenericUtils.collapseSpaces(s);
```

Comment:
I could imagine that there may be pathological cases involving quoted values where this is not the right thing to do; but as far as I see this HostConfigEntry doesn't handle quoted values at all.

Reply:
I see, this is re-used for various types of config file values.  In the case I'm trying to fix it's limited to `Host` only, but I suppose the problem would be generic to any kind of config that supports multiple values (not sure how many of those exist)

I could change the PR to work inside the `if (HOST_CONFIG_PROP.equalsIgnoreCase(key)) {` block, but it would work differently, at this point it would have to filter out empty string values from the list that was parsed incorrectly

Reply:
In searching the `ssh_config` man page for "multiple" and "Multiple" there's quite a few directives that support multiple values, but how they are specified is all over the place. Some are CSV's, some require repeating the directive, some say it supports multiple values but fails to specify how to do so.  

`Include`, for example, contains path-like values and can support multiple values, but it fails to saw how to accomplish this (I'm going to assume multiple `Include` directives, but, it's not explicitly stated like `IdentityFile`)

So it is probably prudent to make this fix in the context of `Host` only to avoid any possible conflict with any other directive-specific